test/backend/inference_backend.py

```diff
@@ -92,12 +95,14 @@ def get_test_models():
         # Adam
 
         # ==OP== Add
+        # ==MIN== 14
```

Comment:
@hamptonm1 A quick reading of the specs seems to indicate that we support add.6, and in fact we may not support add.14's bfloat16?

See https://onnx.ai/onnx/operators/text_diff_Add_6_14.html for differences.

Reply:
@AlexandreEichenberger According to this PR that was merged: https://github.com/onnx/onnx-mlir/pull/1926, we have support for `bfloat16`. In fact, if we look at the documentation `onnx-mlir/src/Dialect/ONNX/ONNXOps.td.inc` it now says we have support for `TensorOf<[BF16]>]>:$C)`. I am assuming this documentation would not be updated unless `onnx-mlir/utils/gen_onnx_mlir.py` was updated with the corresponding opset number. The current version in `onnx-mlir/utils/gen_onnx_mlir.py` for the `Add` operation is 14. 

 Also when I did a print of these two tests (the trick Tong showed me). It says these tests support version 14:
 
    ```
        # ==OP== Add
        # ==LIM== No support for short integers.
        "test_add_cpu": {STATIC_SHAPE:{}, DYNAMIC_SHAPE:{-1:{-1}}, CONSTANT_INPUT:{-1}},
        "test_add_bcast_cpu": {STATIC_SHAPE:{}, DYNAMIC_SHAPE:{-1:{-1}}, CONSTANT_INPUT:{-1}},
```

```
[10/10] cd /Users/meganhampton/zDLC/onnx-mlir/build/test/backend && ONNX_...n3.11 /Users/meganhampton/zDLC/onnx-mlir/build/test/backend/Debug/test.py
opset number: 13
opset number: 7
opset number: 7
opset number: 9
opset number: 9
**opset number: 14
opset number: 14**
```

Reply:
Also we have to be careful because `Add.6` supports the following attributes:

```
### Attributes
--
  |  
  | * **axis - INT** :   If set, defines the broadcast dimensions. See doc for details.
  | * **broadcast - INT** (default is '0'):   Pass 1 to enable broadcasting


```

However, these attributes were removed in `Add.7` and up. So I guess I am not sure if we really support version 6.

Reply:
We can mark it in the limitation for the Op. So we say that we support Add 6 and put in the "Limitation" field for the Op we don't support attributes X and Y.

Reply:
I guess we do not have to.... I used `Netron` to analyze `inception-v2-6.onnx` and the attributes are specified and used for `Add.6`. With an `axis =1` and `broadcast=1`. But of course they no longer exist in version 7 and up.

Reply:
Okay that makes sense!

Reply:
> I believe most of our users will have much more recent models than Opset 6... Probably everything below 9 is very very rare.

Agreed. Outside of the modelzoo 9 is the oldest I recall seeing "in the wild" 

>I personally believe it will be less confusing to them to just say "support 7 and onwards".

13 models in the modelzoo are OpSet 6 and we support all of them so I'd go for "6 and onwards". However I'd advise caution with any "general" min statement at the OpSet level. Between ONNX's choice in how to version OpSets vs versions of the Ops themselves and mixed in with that we don't support all Ops in any OpSet version; it'll get easy to have misunderstandings fast. So I vote we keep this page focused more at the Op version level. 

For example, we'd show a min of 1 for `Conv` since it works in inception-v2-6.onnx and Conv 1 was all that was available in OpSet 6. Just because Conv supports it's version 1, doesn't mean we support OpSet 1 in general. Especially since most other Ops do have a version that we would state the minimum as 6 (OpSet 6 updated a lot of "basic" ops). At least that's my 2cents.

Reply:
@cjvolzka that make sense. You can decide 
(1) no default for min, I you want to be super explicit, or
(2) we can redefine the default min as being the floor... no number below this default.

Reply:
I'd go for option 1 then. When we add a new Op, we specify the Min version we first version we coded and/or tested for the Op. If there's no Min, that implies it's unsupported.

Reply:
I think we should have a default min.... because anything lower than opset 3 is not supported. There are several operations that have not changed since opset 1 for instance `Softsign`. It would not make sense to put 1 here for the minimum given anything under 6 is not supported. Having a default min just gives clear indication that the opset is still supported because there has been no changes since version 1. I think the point of this documentation is to indicate the baseline that the opset still works not when the opset was introduced like in `onnx-mlir/utils/gen_onnx_mlir.py`.

Reply:
If no change was made to the version it means it still works for 6,7,8,9 and etc. The new  version just means when changes were introduced and the subsequent delta compared to the newer version.

Reply:
> because anything lower than opset 6 is not supported. There are several operations that have not changed since opset 1 for instance Softsign. It would not make sense to put 1 here for the minimum given anything under 6 is not supported

On this page, we are documenting specifically the version of the Ops we support. So version 1 would indeed be accurate for Ops like `Conv` and `Softsign` for the reason you pointed out. That doesn't mean we support all of Opset 1 as a whole. Just that we support Version 1 of those particular ops. 

onnx-mlir doesn't have any statements at a model's OpSet level as a whole. Theoretically If you have a model that uses Opset 1, as long as it uses only Ops that we support at V1, then it should work. 

A more likely scenario, just because you have a model that's OpSet 9, it doesn't mean we support it either. We only support it if the model only uses Ops that we document we support. 

> If not, we are frozen at the opset version right BEFORE. The new version just means when changes were introduced and the subsequent delta compared to the newer version.

Correct but again, this is at the individual Op level, not the OpSet as a whole. 

Say we didn't support `Equal` Op version 19. Your model using Opset 19 will work fine with onnx-mlir as long as it doesn't use the Equal Op. If it does use Equal, then you'd have to export the model to a max of 18 (ie OpSet right before the new Equal 19).

Right now we can't accurately make a statement like "OpSet 6 or greater" or "No later than OpSet 18". The only thing we can do accurate document the Ops Versions we do support. From there the user will have to figure out what is the min/max opset they could export their model to and be able to expect it to work with onnx-mlir.

Reply:
Again we need to have a conversation... I do not think you are following. We can talk at the meeting later. Thanks!